canonical-web-and-design/maas-ui
# Fetch license keys

## Done
* Add a license keys nav item.
* Add a basic component which fetches and displays a list of keys.

## QA
A bit tricky, sorry.

1. Ensure you have an amd64 ubuntu image downloaded (via http://maas.local:5240/MAAS/#/images), this is required to populate architecture for the following steps.
2. Create a fake windows image with `dd if=/dev/zero of=windows-dd bs=512 count=10000`
3. Upload the image you've created with `maas {profile} boot-resources create name=windows/win2012 title="Windows Server 2012" architecture=amd64/generic filetype=ddtgz content@=windows-dd`. If you haven't used the `maas` cli before, you'll need to `maas login` first and create a profile.
4. Go to `http://maas.local:5240/MAAS/settings/`, you should see a "license keys" tab now.
5. Add a license key for your image, in the form XXXXX-XXXXX-XXXXX-XXXXX-XXXXX (^([A-Za-z0-9]{5}-){4}[A-Za-z0-9]{5}$).
6. Visit `/MAAS/settings/license-keys` in maas-ui and you should see your license key listed.
